aptos-move/framework/aptos-framework/sources/fungible_asset.move

```diff
@@ -208,23 +223,29 @@ module aptos_framework::fungible_asset {
 
     #[view]
     /// Get the current supply from the `metadata` object.
-    public fun supply<T: key>(metadata: Object<T>): Option<u128> acquires Supply {
+    public fun supply<T: key>(metadata: Object<T>): Option<u128> acquires Supply, ConcurrentSupply {
         let metadata_address = object::object_address(&metadata);
         if (exists<Supply>(metadata_address)) {
             let supply = borrow_global<Supply>(metadata_address);
             option::some(supply.current)
+        } else if (exists<ConcurrentSupply>(metadata_address)) {
+            let supply = borrow_global<ConcurrentSupply>(metadata_address);
+            option::some(aggregator_v2::read(&supply.current))
         } else {
             option::none()
         }
     }
 
     #[view]
     /// Get the maximum supply from the `metadata` object.
-    public fun maximum<T: key>(metadata: Object<T>): Option<u128> acquires Supply {
+    public fun maximum<T: key>(metadata: Object<T>): Option<u128> acquires Supply, ConcurrentSupply {
         let metadata_address = object::object_address(&metadata);
         if (exists<Supply>(metadata_address)) {
             let supply = borrow_global<Supply>(metadata_address);
             supply.maximum
+        } else if (exists<ConcurrentSupply>(metadata_address)) {
+            let supply = borrow_global<ConcurrentSupply>(metadata_address);
+            option::some(aggregator_v2::max_value(&supply.current))
```

Comment:
Similar to previous comment, this should return the user-defined maximum, not the MAX_U128

Reply:
aggregator_v2::max_value is function that gets the value set in the aggregator, not MAX_U128.

though there is one logical change here I missed - if user passed None, we have set MAX_U128 on the aggregator, and would return MAX_U128 here. I can document that - if you think is reasonable, or think of how to return None otherwise

Reply:
Yes some documentation would be useful